src/cluster.c

```diff
@@ -1874,6 +1896,15 @@ int clusterProcessPacket(clusterLink *link) {
      * because of Pub/Sub. */
     if (sender) sender->data_received = now;
 
+    /*
+     * We know the sender node but haven't associate it with the link. This must
+     * be an inbound link because only for inbound links we didn't know which node
+     * to associate when they were created.
+     */
+    if (sender && link->node == NULL) {
+        setClusterNodeToInboundClusterLink(sender, link);
```

Comment:
Once the link->inbound_node has been set we can use that instead of doing the dictionary lookup each time. Seems like it should go up there.

Reply:
Good point. Changed.